src/schnorr.rs

```diff
@@ -86,6 +86,9 @@ impl Signature {
         }
     }
 
+    /// Returns a signature as a byte array.
+    pub fn serialize<T: AsRef<[u8]>>(self) -> [u8; constants::SCHNORR_SIGNATURE_SIZE] { self.0 }
```

Comment:
@Kixunil is this what you meant by adding the `AsRef`? I haven't worked as much with generics yet so want to make sure I got that right

Reply:
No, I meant that in the future we may add `impl AsRef<[u8]> for Signature` and if we do the code using `.as_ref` may become ambiguous. The method helps disambiguate.

So this whole generic is not needed (surprised compiler didn't complain), just delete the entire generic.

Also stylistically I think `&self` is better since that's what most `serialize` methods are taking. But `self` is not strictly wrong.
And it occurred to me that you could put `#[inline]` on top of the method.